src/main/kotlin/org/rust/ide/inspections/RsAssignToImmutableInspection.kt

```diff
@@ -31,20 +33,25 @@ class RsAssignToImmutableInspection : RsLocalInspectionTool() {
         if (left.isMutable) return
 
         when (left) {
-            is RsDotExpr -> registerProblem(holder, expr, "field of immutable binding")
-            is RsIndexExpr -> registerProblem(holder, expr, "indexed content of immutable binding")
+            is RsDotExpr -> registerProblem(holder, expr, left.expr, "field of immutable binding")
+            is RsIndexExpr -> registerProblem(holder, expr, left.containerExpr, "indexed content of immutable binding")
             is RsUnaryExpr -> if (left.isDereference) registerDereferenceProblem(left, holder, expr)
         }
     }
 
     private fun registerDereferenceProblem(left: RsUnaryExpr, holder: ProblemsHolder, expr: RsBinaryExpr) {
         when (left.expr?.type) {
-            is TyReference -> registerProblem(holder, expr, "immutable borrowed content")
-            is TyPointer -> registerProblem(holder, expr, "immutable dereference of raw pointer")
+            is TyReference -> registerProblem(holder, expr, message = "immutable borrowed content")
+            is TyPointer -> registerProblem(holder, expr, message = "immutable dereference of raw pointer")
         }
     }
 
-    private fun registerProblem(holder: ProblemsHolder, expr: RsExpr, message: String) {
-        RsDiagnostic.CannotAssignToImmutable(expr, message).addToHolder(holder)
+    private fun registerProblem(holder: ProblemsHolder, expr: RsExpr, assigneeExpr: RsExpr? = null, message: String) {
```

Comment:
Let's swap `assigneeExpr` and `message` params not to force using named parameter for `message`
If you think that it's important (I have light feel that it's prettier) to keep `expr` and `assigneeExpr` together, you can move `message` param before `expr` one